utilities/zookeeper/R/correctors.county.R

```diff
@@ -0,0 +1,171 @@
+                                 sig_consec = 2.25,
+                                 time_value_flag_date = Sys.Date() + 1,
+                                 multinomial_preprocessor = TRUE,
+                                 corrections_db_path = NULL,
+                                 integer_tol = 1e-6,
+                                 ...) {
+  list(
+    signals_to_correct = signals_to_correct,
+    window_size = window_size,
+    backfill_lag = backfill_lag,
+    excess_cut = excess_cut,
+    size_cut = size_cut,
+    sig_cut = sig_cut,
+    sig_consec = sig_consec,
+    time_value_flag_date = time_value_flag_date,
+    multinomial_preprocessor = multinomial_preprocessor,
+    integer_tol = integer_tol
+  )
+}
+
+#' Corrections function for aardvark
+#'
+#' @param signals_list list of signals as returned from `covidcast_signals()`
+#'   uses the list (no aggregation)
+#' @param ... named corrections parameters passed to `default_county_params()`
+#'
+#' @return a list of signals the same length and format as the input with
+#'   updated entries in the `value` column
+#'
+#'   If the parameter `corrections_db_path` gives a path, intermediate output
+#'   is written to that sqlite database
+#' @export
+zyzzyva_county_corrections <- function(signals_list, ...){
+  if (class(signals_list)[1] == "covidcast_signal") {
+    signals_list <- list(signals_list)
+  }
+  
+  params <- default_county_params(...)
+  len_params <- sapply(params, length)
+  max_len_params <- max(len_params)
+  in_names <- names(signals_list[[1]])
+  assert_that(all(len_params %in% c(1L, max_len_params)),
+              msg = paste("In apply_corrections: ",
+                          "corrections parameters must be length 1 or the",
+                          "same length as the longest corrections parameter."))
+  #Single Signal
+  if (params$signals_to_correct == "response") {
+    corrected <- zyzzyva_county_corrections_single_signal(
+      signals_list[[1]], params)
+    signals_list[[1]] <- corrected %>%
+      mutate(value = .data$corrected) %>%
+      select(all_of(in_names))
+  }
+  
+  # Multiple Signals
+  if (params$signals_to_correct == "all") {
+    corrected <- list()
+    for (i in seq_along(signals_list)) {
+      corrected[[i]] <- zyzzyva_county_corrections_single_signal(
+        signals_list[[i]], params)
+      signals_list[[i]] <- corrected[[i]] %>%
+        mutate(value = .data$corrected) %>%
+        select(all_of(in_names))
+    }
+  }
+  
+  if (!is.null(params$corrections_db_path)) {
+    corrected_df <- bind_rows(corrected) %>%
+      select(.data$data_source, .data$signal, .data$geo_value, .data$time_value,
+             .data$value, .data$corrected, .data$flag)
+    # save it to the db
+    update_corrections(params$corrections_db_path, "county", corrected_df)
+  }
+  
+  return(signals_list)
+}
+
+
+zyzzyva_county_corrections_single_signal <- function(x, params) {
+  if (x == "confirmed_incidence_num") {
```

Comment:
Does this `if` actually work? `x` is a data_frame right.

Reply:
Thank you for reviewing! I will revise it.